File: app/src/main/java/com/stathis/elmepaunivapp/ui/students/recycler/UsefulLinksViewHolder.java
package com.stathis.elmepaunivapp.ui.students.recycler;

import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.stathis.elmepaunivapp.R;
import com.stathis.elmepaunivapp.listeners.UsefulLinkClickListener;
import com.stathis.elmepaunivapp.ui.students.model.UsefulLinks;

public class UsefulLinksViewHolder extends RecyclerView.ViewHolder {

    private UsefulLinkClickListener ItemClickListener;
    private TextView name;
    private ImageView imageView;
    private Object data;

    public UsefulLinksViewHolder(@NonNull View itemView, final UsefulLinkClickListener itemClickListener) {
        super(itemView);

        imageView = itemView.findViewById(R.id.links_img);
        name = itemView.findViewById(R.id.links_holder_txt);

        ItemClickListener = itemClickListener;
        itemView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                ItemClickListener.onUsefulLinksClick((UsefulLinks) data);
            }
        });
    }

    public void present(UsefulLinks data) {
        this.data = data;
        name.setText(data.getName());
        imageView.setImageResource(data.getImageResource());
    }
}

File: app/src/main/java/com/stathis/elmepaunivapp/ui/main/students/recycler/UsefulLinksViewHolder.kt
package com.stathis.elmepaunivapp.ui.main.students.recycler

import androidx.databinding.ViewDataBinding
import com.stathis.elmepaunivapp.abstraction.AbstractViewHolder
import com.stathis.elmepaunivapp.BR
import com.stathis.elmepaunivapp.callbacks.ElmepaClickListener
import com.stathis.elmepaunivapp.abstraction.LocalModel
import com.stathis.elmepaunivapp.ui.main.students.model.UsefulLinks
import com.stathis.elmepaunivapp.ui.main.students.model.LinkItem
import com.stathis.elmepaunivapp.ui.main.students.model.SyllabusItem

class UsefulLinksViewHolder(val binding : ViewDataBinding, val callback: ElmepaClickListener) : AbstractViewHolder(binding) {

    override fun present(data: LocalModel) {
        when (data) {
            is LinkItem -> {
                binding.setVariable(BR.link,data)
                binding.setVariable(BR.callback,callback)
            }

            is SyllabusItem -> {
                binding.setVariable(BR.item,data)
                binding.setVariable(BR.callback,callback)
            }

            is UsefulLinks -> {
                binding.setVariable(BR.usefulLink,data)
                binding.setVariable(BR.callback,callback)
            }
        }
    }
}
